toolkit/offchain/src/reserve/init.rs

```diff
@@ -271,3 +276,44 @@ fn match_costs(
 		Err(anyhow!("Could not build transaction to submit, evaluate response has wrong number of mint keys."))
 	}
 }
+
+// There exist UTXO at Version Oracle Validator with Datum that contains
+// * script id of the script being initialized
+// * Version Oracle Policy Id
+async fn script_is_initialized<
+	T: QueryLedgerState + Transactions + QueryNetwork + QueryUtxoByUtxoId,
+>(
+	script: &ScriptData,
+	version_oracle: &VersionOracleData,
+	ctx: &TransactionContext,
+	client: &T,
+) -> Result<bool, anyhow::Error> {
+	let validator_address = version_oracle.validator.address(ctx.network).to_bech32(None)?;
+	let validator_utxos = client.query_utxos(&[validator_address]).await?;
+	// Decode datum from utxos and check if it contains script id
+	Ok(validator_utxos.into_iter().any(|utxo| {
+		utxo.datum
+			.map(|d| d.bytes)
+			.and_then(|bytes| PlutusData::from_bytes(bytes).ok())
+			.and_then(decode_version_oracle_validator_datum)
+			.filter(|datum| {
+				datum.script_id == script.id
+					&& datum.version_oracle_policy_id == version_oracle.policy_id().0
+			})
+			.is_some()
+	}))
+}
+
+pub(crate) struct VersionOracleValidatorDatum {
+	pub(crate) script_id: u32,
+	pub(crate) version_oracle_policy_id: [u8; 28],
+}
+
+fn decode_version_oracle_validator_datum(data: PlutusData) -> Option<VersionOracleValidatorDatum> {
```

Comment:
Wouldn't it be nicer to put this in a `impl TryFrom<PlutusData> for VersionOracleValidatorDatum`?

Reply:
I would have to write code that deal with many error source only to discard error in the end. If there was `MaybeFrom`